dev-java/saxon/saxon-9.2.0.6-r5.ebuild

```diff
@@ -0,0 +1,67 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+JAVA_PKG_IUSE="doc source"
+
+inherit java-pkg-2 java-pkg-simple
+
+DESCRIPTION="A XSLT and XQuery Processor"
+HOMEPAGE="https://www.saxonica.com/index.html https://saxon.sourceforge.net/"
+SRC_URI="https://downloads.sourceforge.net/saxon/Saxon-HE/$(ver_cut 1-2)/saxonhe${PV//./-}source.zip"
+
+LICENSE="MPL-1.0"
+SLOT="9"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
+IUSE="ant-task"
+
+BDEPEND="app-arch/unzip"
+CP_DEPEND="
+	dev-java/dom4j:1
+	dev-java/jdom:0
+	dev-java/xom:0
+"
+DEPEND="${CP_DEPEND}
+	>=virtual/jdk-1.8:*
+	ant-task? ( >=dev-java/ant-1.10.14-r3:0 )"
```

Comment:
hum, i wonder what is the way to get list only of packages that are either unconditional or for which the use flag is enabled. there sure must be a way, but i have no idea whether that can be done from an ebuild.

Reply:
> hum, i wonder what is the way to get list only of packages that are either unconditional or for which the use flag is enabled. there sure must be a way, but i have no idea whether that can be done from an ebuild.

at the cost of an additional line in the ebuild we could add `COMMON_DEPEND` with that conditional dependency.

but i think it's not worth to spend time on it. we had the same in `dev-java/javacup`. there of course the empty `CP_DEPEND` should be replaced with `COMMON_DEPEND`.

Reply:
i was rather thinking whether we can extend `CP_DEPEND` to work correctly with conditionals.

Reply:
> i was rather thinking whether we can extend `CP_DEPEND` to work correctly with conditionals.

i was having the same idea and that's the reason why i opened https://bugs.gentoo.org/899990. but i don't think it should be a showstopper for this PR. it could be handled separately by someone who understands that coding and has an idea how to improve.

Reply:
yes, you're right, i was just wondering whether there is a simple solution that we could use and improve it right now.